src/lib/features/feature-lifecycle/feature-lifecycle-service.ts

```diff
@@ -169,6 +178,137 @@ export class FeatureLifecycleService {
         }
     }
 
+    private async handleBulkMetrics(events: IClientMetricsEnv[]) {
+        try {
+            const { environments, allFeatures } =
+                this.extractUniqueEnvironmentsAndFeatures(events);
+            const envMap = await this.buildEnvironmentMap();
+            const featureEnvMap =
+                await this.buildFeatureEnvironmentMap(allFeatures);
+            const allStagesToInsert = this.determineLifecycleStages(
```

Comment:
Do we need to do it on handleBulkMetrics? Can we do it using a job that executes on a schedule? I'm thinking that if you have 1000 clients sending metrics this would be expensive and I think we should keep this calculation off the critical path.